iOS-DOCS.md

```diff
@@ -0,0 +1,393 @@
+        * [`onOptOut`](#onoptout)
+        * [`onSkipCardShown`](#onskipcardshown)
+        * [`onUserCancel`](#onusercancel)
+
+## Overview
+
+In order to support interactive ads on iOS, true[X] has created a renderer library that can renderer true[X] ads, which interfaces with a hosting app, as well as its existing ad server and content delivery mechanism (e.g. SSAI).
+
+With this library, the host player app can defer to the TruexAdRenderer when it is required to display a true[X] ad.
+
+For simplicity, publisher implemented code will be referred to as "app code" while true[X] implemented code will be referred to as "renderer code".
+
+true[X] will provide an Objective-C `TruexAdRenderer` library that can be loaded into the app. This library will offer a class, `TruexAdRenderer`, that will need to be instantiated, initialized and given certain commands (described below in [TruexAdRenderer Methods](#truexadrenderer-methods)) by the app code. It will also contain a singleton of shared constants, `TruexConstants`.
+
+At this point, the renderer code will take on the responsibility of requesting ads from true[X] server, creating the UI for the true[X] choice card and interactive ad unit, as well as communicating events to the app code when action is required.
+
+The app code will still need to parse out the SSAI ad response, detect when a true[X] ad is supposed to display, pause the stream, instantiate `TruexAdRenderer` and handle any events emitted by the renderer code. It will also need to call pause, resume and stop methods on the `TruexAdRenderer` when certain external events happen, like if the app is backgrounded or if the user has requested to exit the requested stream via back buttons.
+
+It will also need to handle skipping ads in the current ad pod, if it is notified to do so.
+
+
+## Reference App
+
+**See**: https://github.com/socialvibe/truex-ios-reference-app
+
+We've created a [sample integration](https://github.com/socialvibe/truex-ios-reference-app) which demonstrates a simple intergration with the `TruexAdRenderer` in a media app.
+
+![reference app](media/reference_app.png)
+
+## Product Flows
+
+There are two distinct product flows supported by `TruexAdRenderer`: Sponsored Stream (full-stream ad-replacement) and Sponsored Ad Break (mid-roll ad-replacement).
+
+In a Sponsored Ad Break flow, once the user hits a mid-roll break with a true[X] tag flighted, they will be shown a "choice-card" offering them the choice between watching a normal set of video ads or a fully interactive true[X] ad:
+
+![choice card](media/choice_card.png)
+
+_**Fig. A** example true[X] mid-roll choice card_
+
+If the user opts for a normal ad break, or if the user does not make a selection before the countdown timer expires, the true[X] UI will close and playback of normal video ads can continue as usual.
+
+If the user opts to interact with true[X], an interactive ad unit will be shown to the user:
+
+![ad](media/ad.png)
```

Comment:
I guess tvOS / iOS has not migrated to the footer approach?
Will it eventually?

Reply:
No, tvOS doesn't have a webview and thus, has its native implementation. 
This is more for iOS, which has C2.